geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/IngestCommand.scala

```diff
@@ -31,58 +34,77 @@ trait IngestCommand[DS <: DataStore] extends DataStoreCommand[DS] {
   def libjarsPaths: Iterator[() => Seq[File]]
 
   override def execute(): Unit = {
-    import DataFormats.{Avro, Csv, Shp, Tsv}
+    import DataFormats._
+    import org.locationtech.geomesa.utils.conversions.ScalaImplicits.RichIterator
 
     ensureSameFs(PathUtils.RemotePrefixes)
 
-    val ingest = if (params.fmt == Shp) {
-      createShpIngest()
-    } else if (params.spec == null && params.config == null && Seq(Csv, Tsv, Avro).contains(params.fmt)) {
-      // if there is no sft and no converter passed in, try to use the auto ingest which will
-      // pick up the schema from the input files themselves
-      if (params.featureName == null) {
-        throw new ParameterException("Feature name is required when a schema is not specified")
-      }
-      // auto-detect the import schema
-      Command.user.info("No schema or converter defined - will attempt to detect schema from input files")
-      createAutoIngest()
+    if (params.fmt == Shp) {
+      createShpIngest().run()
     } else {
-      // validate arguments
-      if (params.config == null) {
-        throw new ParameterException("Converter Config argument is required")
-      }
-      val converterConfig = CLArgResolver.getConfig(params.config)
-
-      val sft = if (params.spec != null) {
-        CLArgResolver.getSft(params.spec, params.featureName)
-      } else if (params.featureName != null) {
-        Try(withDataStore(_.getSchema(params.featureName))).filter(_ != null).getOrElse {
-          throw new ParameterException(s"SimpleFeatureType '${params.featureName}' does not currently exist, " +
-              "please provide specification argument")
+      // try to load the sft, first check for an existing schema, then load from the params/environment
+      var sft: SimpleFeatureType =
+        Option(params.featureName).flatMap(n => Try(withDataStore(_.getSchema(n))).filter(_ != null).toOption)
+          .orElse(Option(params.spec).flatMap(s => Option(CLArgResolver.getSft(s, params.featureName))))
+          .orNull
+
+      var converter: Config = Option(params.config).map(CLArgResolver.getConfig).orNull
+
+      if (converter == null && params.files.nonEmpty) {
+        // if there is no converter passed in, try to infer the schema from the input files themselves
```

Comment:
What happens if a user has a bunch of files of the same format, ingests some and then ingests some more?  

Does the inferencing re-run?  Or is the SFT/converter definition saved after the first run?

Reply:
the command logs out to the user that you need to save the converter definition if you want to ingest with it again. you can re-run without doing that, but the converter might not end up the same...